Patches/Rimworld - Altered Carbon/ThingDefs_Weapons.xml

```diff
@@ -0,0 +1,366 @@
+            <li Class="CombatExtended.GunDrawExtension">
+              <DrawSize>0.65,0.65</DrawSize>
+              <DrawOffset>0.0,-0.02</DrawOffset>
+            </li>
+          </value>
+        </li>
+
+        <!-- === MAC Revolver === -->
+        <li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+          <defName>AC_Gun_MACRevolver</defName>
+
+          <statBases>
+            <Mass>1.89</Mass>
+            <Bulk>2.80</Bulk>
+            <SwayFactor>1.20</SwayFactor>
+            <ShotSpread>0.16</ShotSpread>
+            <SightsEfficiency>1.1</SightsEfficiency>
+            <RangedWeapon_Cooldown>0.45</RangedWeapon_Cooldown>
+          </statBases>
+
+          <Properties>
+            <verbClass>CombatExtended.Verb_ShootCE</verbClass>
+            <hasStandardCommand>True</hasStandardCommand>
+            <defaultProjectile>Bullet_8x35mmCharged</defaultProjectile>
+            <warmupTime>0.6</warmupTime>
+            <range>12</range>
+            <soundCast>Shot_Revolver</soundCast>
+            <soundCastTail>GunTail_Medium</soundCastTail>
+            <muzzleFlashScale>9</muzzleFlashScale>
+          </Properties>
+
+          <AmmoUser>
+            <magazineSize>10</magazineSize>
+            <reloadTime>4.5</reloadTime>
+            <ammoSet>AmmoSet_8x35mmCharged</ammoSet>
+          </AmmoUser>
+
+          <FireModes>
+            <aiAimMode>AimedShot</aiAimMode>
+          </FireModes>
+
+          <weaponTags>
+            <li>SpacerGun</li>
+            <li>AC_LateIndustrialGun</li>
+            <li>CE_Sidearm</li>
+            <li>CE_OneHandedWeapon</li>
+          </weaponTags>
+        </li>
+
+        <!-- === GunDrawExtension === -->
+        <li Class="PatchOperationAddModExtension">
+          <xpath>/Defs/ThingDef[defName="AC_Gun_MACRevolver"]</xpath>
+
+          <value>
+            <li Class="CombatExtended.GunDrawExtension">
+              <DrawSize>0.80,0.80</DrawSize>
+              <DrawOffset>0.0,-0.08</DrawOffset>
+            </li>
+          </value>
+        </li>
+
+        <!-- === Shock PDW === -->
+        <li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+          <defName>AC_Gun_ShockPDW</defName>
+
+          <statBases>
+            <Mass>3.2</Mass>
+            <Bulk>5.00</Bulk>
+            <SwayFactor>1.04</SwayFactor>
+            <ShotSpread>0.09</ShotSpread>
+            <SightsEfficiency>2</SightsEfficiency>
+            <RangedWeapon_Cooldown>0.36</RangedWeapon_Cooldown>
+          </statBases>
+
+          <Properties>
+            <recoilAmount>1.30</recoilAmount>
+            <verbClass>CombatExtended.Verb_ShootCE</verbClass>
+            <hasStandardCommand>True</hasStandardCommand>
+            <defaultProjectile>Bullet_6x24mmCharged</defaultProjectile>
+            <burstShotCount>6</burstShotCount>
+            <ticksBetweenBurstShots>6</ticksBetweenBurstShots>
+            <warmupTime>0.6</warmupTime>
+            <range>46</range>
```

Comment:
This seems pretty high for a PDW (for reference, the P90 has a range of 20). Is this gun a PDW, or a carbine?

Reply:
It's basically a modernized AKS-74U and the reason i gave it more range it's because lore-wise it uses an experimental railgun system, just to make it different idk seemed fitting.